src/main/java/duke/component/Parser.java

```diff
@@ -0,0 +1,89 @@
+package duke.component;
+
+import duke.command.*;
+
+import duke.exception.EmptyDescriptionException;
+import duke.exception.UnknownCommandException;
+import duke.exception.WrongFormatException;
+import duke.task.ToDo;
+import duke.task.Deadline;
+import duke.task.Event;
+
+public class Parser {
+    /**
+     * Parses command and generate task accordingly.
+     * @param command
+     * @return
+     * @throws UnknownCommandException
+     * @throws EmptyDescriptionException
+     * @throws WrongFormatException
+     */
+    public static Command parse(String command) throws UnknownCommandException, EmptyDescriptionException, WrongFormatException {
+        String[] parameters = command.split(" ");
+        switch (parameters[0]) {
+        case "bye": {
+            return new ExitCommand();
+        }
+        case "list": {
+            return new ListCommand();
+        }
+        case "done": {
+            if (command.length() <= 5) {
```

Comment:
I noticed a lot of command.length() checkers in if statements across many case blocks. Maybe abstracting them out into a single method could help reduce indentation levels and case block length?